File: mobile/ios/ShiftPlanner/ShiftPlanner/Features/Manager/Requirements/Repositories/RequirementsRepository.swift
import Foundation

protocol RequirementsRepository {
    func fetchPositions() async throws -> [RequirementPositionOption]
    func fetchRequirements(startDate: Date, endDate: Date) async throws -> [RequirementOccurrence]
    func createRequirement(
        date: Date,
        branchId: Int?,
        positionId: Int,
        quantity: Int,
        startSlot: Int,
        endSlot: Int
    ) async throws -> RequirementOccurrence
    func updateRequirement(
        id: Int,
        date: Date,
        branchId: Int?,
        positionId: Int,
        quantity: Int,
        startSlot: Int,
        endSlot: Int
    ) async throws -> RequirementOccurrence
    func createRequirementsBulk(
        startDate: Date,
        endDate: Date,
        weekdays: [Int],
        templates: [RequirementTemplateDraft]
    ) async throws -> [RequirementOccurrence]
    func deleteRequirement(id: Int) async throws
}

File: mobile/android/app/src/main/java/com/froggyriia/shiftplanner/data/requirements/RequirementsRepository.kt
package com.froggyriia.shiftplanner.data.requirements

import com.froggyriia.shiftplanner.domain.model.RequirementOccurrence
import com.froggyriia.shiftplanner.domain.model.RequirementPositionOption
import com.froggyriia.shiftplanner.domain.model.RequirementTemplateDraft

interface RequirementsRepository {
    suspend fun fetchPositions(): List<RequirementPositionOption>
    suspend fun fetchRequirements(startDate: String, endDate: String): List<RequirementOccurrence>
    suspend fun createRequirement(
        date: String,
        branchId: Int?,
        positionId: Int,
        quantity: Int,
        startSlot: Int,
        endSlot: Int
    ): RequirementOccurrence
    suspend fun updateRequirement(
        id: Int,
        date: String,
        branchId: Int?,
        positionId: Int,
        quantity: Int,
        startSlot: Int,
        endSlot: Int
    ): RequirementOccurrence
    suspend fun deleteRequirement(id: Int)
    suspend fun importRequirementsXlsx(fileBytes: ByteArray, fileName: String): RequirementsImportResult
    suspend fun createRequirementsBulk(
        startDate: String,
        endDate: String,
        weekdays: List<Int>,
        templates: List<RequirementTemplateDraft>
    ): List<RequirementOccurrence>
}
